### PR TITLE
[DOC] Add docs of Runtime Filter

### PR DESCRIPTION
**Pull request by xinyiZzz**
## Proposed changes

Add docs of Runtime Filter #6116

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation Update (if none of the other choices apply)
- [ ] Code refactor (Modify the code structure, format the code, etc...)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I have created an issue on (Fix #ISSUE) and described the bug/feature there in detail
- [ ] Compiling and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If these changes need document changes, I have updated the document
- [ ] Any dependent changes have been merged

### CODE REVIEW

File: docs/en/administrator-guide/runtime-filter.md [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+RuntimeFilter:in:
+    -  HasPushDownToEngine:  true
+    -  AWaitTimeCost:  0ns
+    -  EffectTimeCost:  2.76ms
+
+-- 此外，在profile的OLAP_SCAN_NODE中还可以查看Runtime Filter下推后的过滤效果和耗时。
+    -  RowsVectorPredFiltered:  9.320008M  (9320008)
+    -  VectorPredEvalTime:  364.39ms
+```
+
```
Comment (hf200012) [inline docs/en/administrator-guide/runtime-filter.md:209]: It should be English here

Comment (xinyiZzz) [reply docs/en/administrator-guide/runtime-filter.md]: Modified, thks

File: docs/zh-CN/administrator-guide/runtime-filter.md [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+|        T1                        T2
+|
+```
+可见，和谓词下推、分区裁剪不同，Runtime Filter是在运行时动态生成的过滤条件，即在查询运行时解析join on clause确定过滤表达式，并将表达式广播给正在读取左表的ScanNode，从而减少扫描的数据量，进而减少probe hash table的次数，避免不必要的I/O和网络传输。
+
+Runtime Filter主要用于优化针对大表的join，如果左表的数据量太小，或者右表的数据量太大，则Runtime Filter可能不会取得预期效果。
+
+## 使用方式
+
+### 设置Session变量
```
Comment (morningman) [inline docs/zh-CN/administrator-guide/runtime-filter.md:85]: 建议在这里先把所有runtime filter相关参数罗列一些，然后说明，在绝大多数情况下，只需要修改 runtime_filter_type，其他参数保持默认即可。在某些特定场景下，才需要进一步的调整其他参数已达到最优效果

**Review by morningman — COMMENTED** [review]
You need to add these doc to `docs/.vuepress/sidebar/*.js`

File: docs/en/administrator-guide/runtime-filter.md [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+* Left table: the table on the left during Join query. Perform Probe operation. The order can be adjusted by Join Reorder.
+* Right table: the table on the right during Join query. Perform the Build operation. The order can be adjusted by Join Reorder.
+* Fragment: FE will convert the execution of specific SQL statements into corresponding fragments and send them to BE for execution. The corresponding Fragment is executed on the BE, and the results are aggregated and returned to the FE.
+* Join on clause: `Aa=Bb` in `A join B on Aa=Bb`, based on this to generate join conjuncts during query planning, including expr used by join Build and Probe, where Build expr is called in Runtime Filter src expr, Probe expr are called target expr in Runtime Filter.
+
+## Principle
+Runtime Filter is generated during query planning, constructed in HashJoinNode, and applied in ScanNode.
+
+For example, there is currently a Join query between the T1 table and the T2 table. Its Join mode is HashJoin. T1 is a fact table with 100,000 rows of data. T2 is a dimension table with 100 rows of data. Doris join The actual situation is:
+```
```
Comment (pengxiangyu) [inline docs/en/administrator-guide/runtime-filter.md:44]: 这里写100 rows，下面图里为什么是2000?

File: docs/zh-CN/administrator-guide/runtime-filter.md [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+* BE：Backend，Doris 的后端节点。负责查询执行和数据存储。
+* 左表：Join查询时，左边的表。进行Probe操作。可被Join Reorder调整顺序。
+* 右表：Join查询时，右边的表。进行Build操作。可被Join Reorder调整顺序。
+* Fragment：FE会将具体的SQL语句的执行转化为对应的Fragment并下发到BE进行执行。BE上执行对应Fragment，并将结果汇聚返回给FE。
+* Join on clause: `A join B on A.a=B.b`中的`A.a=B.b`，在查询规划时基于此生成join conjuncts，包含join Build和Probe使用的expr，其中Build expr在Runtime Filter中称为src expr，Probe expr在Runtime Filter中称为target expr。
+
+## 原理
+Runtime Filter在查询规划时生成，在HashJoinNode中构建，在ScanNode中应用。
+
+举个例子，当前存在T1表与T2表的Join查询，它的Join方式为HashJoin，T1是一张事实表，数据行数为100000，T2是一张维度表，数据行数为100，Doris join的实际情况是:
```
Comment (pengxiangyu) [inline docs/zh-CN/administrator-guide/runtime-filter.md:43]: 是100还是2000？

File: docs/zh-CN/administrator-guide/runtime-filter.md [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+### 设置Session变量
+除`runtime_filter_type`外，其他查询选项用于调整Runtime Filter在特定场景下达到最佳性能，通常只在性能测试后进行调整，针对资源密集型、运行耗时足够长且频率足够高的查询进行优化。
+
+#### 1.runtime_filter_type
+使用的Runtime Filter类型，包括Bloom Filter、MinMax Filter、IN predicate三种类型，使用多个时用逗号分隔，注意需要加引号，例如:
+```
+set runtime_filter_type="BLOOM_FILTER,IN,MIN_MAX";
+```
+默认会保守的只使用IN predicate，部分情况下同时使用Bloom Filter、MinMax Filter、IN predicate时性能更高。
+
```
Comment (pengxiangyu) [inline docs/zh-CN/administrator-guide/runtime-filter.md:94]: 分别解释一下Bloom Filter   MinMax Filter   IN predicate的具体算法吧。

File: docs/zh-CN/administrator-guide/runtime-filter.md [hunk 6]
```diff
@@ -0,0 +1,223 @@
...
+- **Bloom Filter**: 有一定的误判率，导致过滤的数据比预期少一点，但不会导致最终结果不准确，在大部分情况下Bloom Filter都可以提升性能或对性能没有显著影响，但在部分情况下会导致性能降低。
+    - Bloom Filter构建和应用的开销较高，所以当过滤率较低时，或者左表数据量较少时，Bloom Filter可能会导致性能降低。
+    - 目前只有左表的Key列应用Bloom Filter才能下推到存储引擎，而测试结果显示Bloom Filter不下推到存储引擎时往往会导致性能降低。
+    - 目前Bloom Filter仅在ScanNode上使用表达式过滤时有短路(short-circuit)逻辑，即当假阳性率过高时，不继续使用Bloom Filter，但当Bloom Filter下推到存储引擎后没有短路逻辑，所以当过滤率较低时可能导致性能降低。
+- **MinMax Filter**: 包含最大值和最小值，从而过滤小于最小值和大于最大值的数据，MinMax Filter的过滤效果与join on clause中Key列的类型和左右表数据分布有关。
+    - 当join on clause中Key列的类型为int/bigint/double等时，极端情况下，如果左右表的最大最小值相同则没有效果，反之右表最大值小于左表最小值，或右表最小值大于左表最大值，则效果最好。
+    - 当join on clause中Key列的类型为varchar等时，应用MinMax Filter往往会导致性能降低。
+- **IN predicate**: 根据join on clause中Key列在右表上的所有值构建IN predicate，使用构建的IN predicate在左表上过滤，相比loom Filter构建和应用的开销更低，在右表数据量较少时往往性能更高。
```
Comment (pengxiangyu) [inline docs/zh-CN/administrator-guide/runtime-filter.md:102]: loom Filter拼写错误

File: docs/zh-CN/administrator-guide/runtime-filter.md [hunk 7]
```diff
@@ -0,0 +1,223 @@
...
+用于调整Runtime Filter的下推策略，包括LOCAL和GLOBAL两种策略，默认设置为GLOBAL策略。
+
+LOCAL：相对保守，构建的Runtime Filter只能在同一个instance（查询执行的最小单元）上同一个Fragment中使用，即Runtime Filter生产者（构建Filter的HashJoinNode）和消费者（使用RuntimeFilter的ScanNode）在同一个Fragment，比如broadcast join的一般场景；
+
+GLOBAL：相对激进，除满足LOCAL策略的场景外，还可以将Runtime Filter合并后通过网络传输到不同instance上的不同Fragment中使用，比如Runtime Filter生产者和消费者在不同Fragment，比如shuffle join，这可以在更广泛的场景对查询进行优化。
+
+在不同Fragment上构建和应用Runtime Filter时，需要合并Runtime Filter的原因和策略可参阅 [ISSUE 6116](https://github.com/apache/incubator-doris/issues/6116)
+
+#### 3.runtime_filter_wait_time_ms
+在开启Runtime Filter后，join左表的ScanNode会为每一个分配给自己的Runtime Filter等待一段时间再扫描数据，默认等待1s（1000ms，以毫秒为单位），即如果ScanNode被分配了3个Runtime Filter，那么它最多会等待3s。
```
Comment (pengxiangyu) [inline docs/zh-CN/administrator-guide/runtime-filter.md:116]: 为什么是3秒？不是并行的么？

File: docs/en/administrator-guide/runtime-filter.md (hunk 3 above)
Comment (xinyiZzz) [reply docs/en/administrator-guide/runtime-filter.md:44]: Modified, thks

File: docs/zh-CN/administrator-guide/runtime-filter.md (hunk 4 above)
Comment (xinyiZzz) [reply docs/zh-CN/administrator-guide/runtime-filter.md:43]: Modified, thks

File: docs/zh-CN/administrator-guide/runtime-filter.md (hunk 5 above)
Comment (xinyiZzz) [reply docs/zh-CN/administrator-guide/runtime-filter.md:94]: 在下面有对这三个filter说明

File: docs/zh-CN/administrator-guide/runtime-filter.md (hunk 6 above)
Comment (xinyiZzz) [reply docs/zh-CN/administrator-guide/runtime-filter.md:102]: Modified, thks

File: docs/zh-CN/administrator-guide/runtime-filter.md (hunk 7 above)
Comment (xinyiZzz) [reply docs/zh-CN/administrator-guide/runtime-filter.md:116]: 不是，目前是在一个循环里面，一个一个等待的

File: docs/zh-CN/administrator-guide/runtime-filter.md (hunk 2 above)
Comment (xinyiZzz) [reply docs/zh-CN/administrator-guide/runtime-filter.md:85]: Modified, thks

File: docs/zh-CN/administrator-guide/runtime-filter.md [hunk 8]
```diff
@@ -0,0 +1,279 @@
...
+**使用注意事项**
+
+在开启Runtime Filter后，左表的ScanNode会为每一个分配给自己的Runtime Filter等待一段时间再扫描数据，即如果ScanNode被分配了3个Runtime Filter，那么它最多会等待3000ms。
+
+因为Runtime Filter的构建和合并均需要时间，ScanNode会尝试将等待时间内到达的Runtime Filter下推到存储引擎，如果超过等待时间后，ScanNode会使用已经到达的Runtime Filter直接开始扫描数据。
+
+如果Runtime Filter在ScanNode开始扫描之后到达，则ScanNode不会将该Runtime Filter下推到存储引擎，而是对已经从存储引擎扫描上来的数据，在ScanNode上基于该Runtime Filter使用表达式过滤，之前已经扫描的数据则不会应用该Runtime Filter，这样得到的中间数据规模会大于最优解，但可以避免严重的裂化。
+
+如果集群比较繁忙，并且集群上有许多资源密集型或长耗时的查询，可以考虑增加等待时间，以避免复杂查询错过优化机会。如果集群负载较轻，并且集群上有许多只需要几秒的小查询，可以考虑减少等待时间，以避免每个查询增加1s的延迟。
+
```
Comment (pengxiangyu) [inline docs/zh-CN/administrator-guide/runtime-filter.md:172]: 错过优化机会是什么意思？如何错过的？如何才能避免错过？

File: docs/zh-CN/administrator-guide/runtime-filter.md [hunk 9]
```diff
@@ -0,0 +1,279 @@
...
+#### 3.runtime_filter_wait_time_ms
+Runtime Filter的等待耗时。
+
+**类型**: 整数，默认1000，单位ms
+
+**使用注意事项**
+
+在开启Runtime Filter后，左表的ScanNode会为每一个分配给自己的Runtime Filter等待一段时间再扫描数据，即如果ScanNode被分配了3个Runtime Filter，那么它最多会等待3000ms。
+
```
Comment (pengxiangyu) [inline docs/zh-CN/administrator-guide/runtime-filter.md:166]: 为什么是3000ms？不是并行的么？

File: docs/zh-CN/administrator-guide/runtime-filter.md (hunk 5 above)
Comment (pengxiangyu) [reply docs/zh-CN/administrator-guide/runtime-filter.md:127]: Bloom Filter如何过滤的，没有说明白。

Comment (xinyiZzz) [reply docs/zh-CN/administrator-guide/runtime-filter.md:127]: Bloom Filter的概念和原理感觉没必要在这里解释了，解释起来繁琐，这应该算基础知识…

File: docs/zh-CN/administrator-guide/runtime-filter.md (hunk 8 above)
Comment (xinyiZzz) [reply docs/zh-CN/administrator-guide/runtime-filter.md:172]: 错过：如果没有在规定时间内等到Runtime filter到达，会不使用Runtime FIlter直接开始扫描，即没用上runtime filter
避免：可以考虑增加等待时间

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**